### PR TITLE
Implement fast hashtable with pruning in C

### PR DESCRIPTION
**Pull request by isamaru**
This is incremental to changes in cbounter which is only about CMS, but provides necessary libraries and infrastructure.

### CODE REVIEW

File: scripts/ht_size_test.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import pickle
+
+from HTC import HT_Basic
```
Comment (menshikh-iv) [inline scripts/ht_size_test.py]: Please use lower case for module names (here and anywhere, in #10 too)

File: scripts/ht_size_test.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import pickle
```
Comment (menshikh-iv) [inline scripts/ht_size_test.py]: Same comments for "test files" as in #10

File: scripts/ht_test.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+print("total %d (expected %d)" % (htc2.total(), 1422))
+print("length %d (expected %d)" % (len(htc2), 5))
+
+del htc2["foo"]
+expected["foo"] = 0
+print("deleted")
+
+print("saving")
+pickle.dump(htc2, open("HTC.p", 'wb'))
```
Comment (menshikh-iv) [inline scripts/ht_test.py]: Please close files after using, you can do it in automatic way with `with` operator

```python
with open("HTC.p", 'wb') as outfile:
    pickle.dump(htc2, outfile)
```

File: scripts/purge_mem_test.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import time
+
+import smart_open
+from HTC import HT_Basic as HashTable
+
+
+def loadWiki(counter, articles=1):
```
Comment (menshikh-iv) [inline scripts/purge_mem_test.py]: For function name please don't use `camelCase`, only lowercase.

File: scripts/purge_mem_test.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    loaded_articles = 0
+    words = 0
+    wiki = smart_open.smart_open('C:/rare/corpus/wiki/title_tokens.txt.gz')
```
Comment (menshikh-iv) [inline scripts/purge_mem_test.py]: `with smart_open.smart_open('C:/rare/corpus/wiki/title_tokens.txt.gz') as wiki`

File: scripts/purge_mem_test.py (hunk 4 above)
Comment (piskvorky) [reply scripts/purge_mem_test.py]: `lowercase_with_underscore` in Python.

**Pull request merged**